### PR TITLE
Whiskey (New)

### PR DESCRIPTION
**Pull request by cite2000**
## About the PR
Adding the NSFD "Whiskey" ship.

## Why / Balance
I felt NFSD needed a heavy rescue ship since the next best medical vessel is the Hospitaller which isn't always the best when there's more than one injured or dead. There are certain situations where you just simple need more medical capabilities to be able to handle of the dead. Some situations this is the case is in some POI like the labs and asteroid which can easily cause the death of entire crews if things go bad. In situations like this the hospitaller is simply not large enough and the empress and other such ships seem like improper due to them being flagships and slow to maneuver especially when other POIs need their attention more such as vaults and syndicate stashes.

## Media
whiskey lower init
whiskey lower litup
whiskey upper init
whiskey upper litup

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: NFSD medical ship "Whiskey"

### CODE REVIEW

**Review by FoxxoTrystan — APPROVED** [review]

File: Resources/Maps/Shuttles/whiskey.yml
Comment (TsjipTsjip) [inline Resources/Maps/Shuttles/whiskey.yml:1]: We should seriously consider whether or not we want this kind of ship in frontier.

Major notes:
- Are these holding cells? 1-tile? This has to be remediated.
image
- Remove the loose headset.
image
- Table through directional window.
image
- Cloning on a non-flagship feels iffy. The Helix having it is already a bit of a stretch, but acceptable as it is _purely_ medical, not a security ship. Also equipment is clipping with the directional windows.
image
- Engineering feels very overkill for what is essentially a shuttle meant to be manned by two brigmedics. Remove the engineering locker for starters, superpacman feels overkill and should be replaced with two regular pacmans. It just feels too over-equipped.
image

Minor notes:
- Doctors should not be smoking. That's just opinion tho.
image

As it stands, this ship is not ready to be merged and needs a bit of work still.

**Review by TsjipTsjip — CHANGES_REQUESTED** [review]
Bunch of notes, this ship also needs a maintainer to give their approval on it before it is merged in the sense of the concept. I'm not sure if we need this ship on the security's ship catalog.

**Comment by TsjipTsjip** [comment]
Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.

File: Resources/Prototypes/_NF/Shipyard/whiskey.yml
Comment (TsjipTsjip) [reply Resources/Prototypes/_NF/Shipyard/whiskey.yml:1]: This is resolved in aa3382172b74815d94c7d0a8eb275acb9e12f037 .

**Comment by YumiGumi** [comment]
> Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.

glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.

**Comment by dvir001** [comment]
> > Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.
> 
> glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.

I agree that glass underneath the thrusters feels very weird, logic wise it should not be a thing and it makes then look disconnected from the shuttle

This is something that should be updated and fixed with older ships who were held to different map standards before.

Were getting more hyper critical as were getting ships who add more mapper maintainers upkeep while having issues that could have been fixed while they were still being worked on, then having someone else fix later.

Look at how many shuttles had to get full atmost redone in the past 2 weeks since this was not a requirement for smalls ships before.

There is a baseline doc coming out soon to help streamline this process to have mapping more clear.

**Comment by YumiGumi** [comment]
> > > Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.
> > 
> > 
> > glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.
> 
> I agree that glass underneath the thrusters feels very weird, logic wise it should not be a thing and it makes then look disconnected from the shuttle
> 
> This is something that should be updated and fixed with older ships who were held to different map standards before.
> 
> Were getting more hyper critical as were getting ships who add more mapper maintainers upkeep while having issues that could have been fixed while they were still being worked on, then having someone else fix later.
> 
> Look at how many shuttles had to get full atmost redone in the past 2 weeks since this was not a requirement for smalls ships before.
> 
> There is a baseline doc coming out soon to help streamline this process to have mapping more clear.

the ship i'm specifically regarding was made by magnus in the last month.

**Comment by dvir001** [comment]
> > > > Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.
> > > 
> > > 
> > > glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.
> > 
> > 
> > I agree that glass underneath the thrusters feels very weird, logic wise it should not be a thing and it makes then look disconnected from the shuttle
> > This is something that should be updated and fixed with older ships who were held to different map standards before.
> > Were getting more hyper critical as were getting ships who add more mapper maintainers upkeep while having issues that could have been fixed while they were still being worked on, then having someone else fix later.
> > Look at how many shuttles had to get full atmost redone in the past 2 weeks since this was not a requirement for smalls ships before.
> > There is a baseline doc coming out soon to help streamline this process to have mapping more clear.
> 
> the ship i'm specifically regarding was made by magnus in the last month.

I dont care who made the ship, if its need to be changed it will.

**Comment by TsjipTsjip** [comment]
> but i'll add you've been hyper critical of the ship since it's conception

The only comment I have made before about this ship on discord is that plastitanium walls felt weird. This was wrong, as I have since learned it's common practice. I am sorry you feel this way about my comments, I should add that I am not a maintainer and as such you're allowed to put all my feedback in the disposal unit if you so choose. If you know better than me when it comes to mapping, then do do that.

Aside from my final review that I'll leave on this PR in a minute or so which will only contain critical points, I am detaching myself from this pull request as I believe any more feedback I give will not be treated receptively.

**Review by TsjipTsjip — CHANGES_REQUESTED** [review]
- Spawning issue is resolved.
- Brigmed headset **has to be removed entirely**, it contains an NFSD key which should only ever be normally obtainable through spawning as a role, getting it from SR/Sheriff in a PTech, or via "illegal" means (i.e. artifacts, ...) (I know there is telecomms servers around, this will be remediated as well in a later PR effort.)

**Comment by dvir001** [comment]
> * Brigmed headset

Ill get the Brigmed headset

**Comment by dvir001** [comment]
Sorry about the close, this is a mistake on my part with the comment click.

**Comment by YumiGumi** [comment]
e

> > > > > Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.
> > > > 
> > > > 
> > > > glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.
> > > 
> > > 
> > > I agree that glass underneath the thrusters feels very weird, logic wise it should not be a thing and it makes then look disconnected from the shuttle
> > > This is something that should be updated and fixed with older ships who were held to different map standards before.
> > > Were getting more hyper critical as were getting ships who add more mapper maintainers upkeep while having issues that could have been fixed while they were still being worked on, then having someone else fix later.
> > > Look at how many shuttles had to get full atmost redone in the past 2 weeks since this was not a requirement for smalls ships before.
> > > There is a baseline doc coming out soon to help streamline this process to have mapping more clear.
> > 
> > 
> > the ship i'm specifically regarding was made by magnus in the last month.
> 
> I dont care who made the ship, if its need to be changed it will.

Right, because a maintainer can do whatever they please and the rest of us get the shaft. The sheer amount of a headache that has become mapping recently has turned me off from wanting to contribute at all. I understand the mapping baselines are in the works at the moment, but don't have one ship be allowed one thing and then change that stance without fixing the lot. It's just getting to the point of if you're a maintainer then anything is free game to you, but when a contributor does something it's met with unfathomable scrutiny and issues. I respect the fact that as a maintainer magnus has a lot more free reign regarding implementation, but when one person follows the example set by said maintainer we get hounded and harassed for it. The fucking glass looks nice, it's transparent and doesn't impact the ship in any form except how you think it should look. Until you figure out what is and isn't acceptable and everyone starts mapping things to that, i think i'll just retire myself from contributing.

**Comment by cite2000** [comment]
> * Spawning issue is resolved.
> * Brigmed headset **has to be removed entirely**, it contains an NFSD key which should only ever be normally obtainable through spawning as a role, getting it from SR/Sheriff in a PTech, or via "illegal" means (i.e. artifacts, ...) (I know there is telecomms servers around, this will be remediated as well in a later PR effort.)

while this is the case will the hospitaller be changed to no longer include a headset with sec comms in it? will update to remove the headset shortly.

**Comment by TsjipTsjip** [comment]
> while this is the case will the hospitaller be changed to no longer include a headset with sec comms in it? will update to remove the headset shortly.

I will add this to the list of things to do. Thanks for flagging it.

**Comment by cite2000** [comment]
> > while this is the case will the hospitaller be changed to no longer include a headset with sec comms in it? will update to remove the headset shortly.
> 
> I will add this to the list of things to do. Thanks for flagging it.

actually on further inspection quite a few sec ships have sec comms in them. every baliff locker has sec comms in it. as does every sheriff locker.

**Comment by TsjipTsjip** [comment]
> actually on further inspection quite a few sec ships have sec comms in them. every baliff locker has sec comms in it. as does every sheriff locker.

If it's in the locker fill that points to an underlying issue... Gonna @dvir001 here cause they know what can and can't be removed from the locker fill without breaking something in the most horrible way imaginable. (Imagine a sheriff without comms!)

**Comment by cite2000** [comment]
> > actually on further inspection quite a few sec ships have sec comms in them. every baliff locker has sec comms in it. as does every sheriff locker.
> 
> If it's in the locker fill that points to an underlying issue... Gonna @dvir001 here cause they know what can and can't be removed from the locker fill without breaking something in the most horrible way imaginable. (Imagine a sheriff without comms!)

though one thing i fail to see the issue in is the generator, it does the exact same thing as a pacman generator but requires more expensive fuel, and i'd have to considerably change things in the ship to fit another generator in there. as it is i am working with quite limited space. and i could change the cells into 1x2 or 2x2 but that would change the layout of the doors.

**Comment by TsjipTsjip** [comment]
> though one thing i fail to see the issue in is the generator, it does the exact same thing as a pacman generator but requires more expensive fuel, and i'd have to considerably change things in the ship to fit another generator in there.

While mostly opinion, an argument can be made that this ship will be bought and sold just for the SUPERPACMAN, as this is usually just expedition loot. I'm not the definitive source on this tho, so feel free to do what you like until a maintainer decides on this.

> and i could change the cells into 1x2 or 2x2 but that would change the layout of the doors. it isn't MEANT to be a prisoner transport, it's a medical vessel.

2x2 would be the very least that's acceptable. (Mostly in the "humane treatment of prisoners" category, we don't want to encourage officers to mistreat prisoners and having good areas for them is part of that.) Alternatively, if you feel this isn't a prisoner transport, I would just remove the holding cells entirely, replacing them with plastitanium walls. Having doubled up walls isn't necessary problematic, especially when there's nothing else you can do with it.

**Comment by cite2000** [comment]
> > though one thing i fail to see the issue in is the generator, it does the exact same thing as a pacman generator but requires more expensive fuel, and i'd have to considerably change things in the ship to fit another generator in there.
> 
> While mostly opinion, an argument can be made that this ship will be bought and sold just for the SUPERPACMAN, as this is usually just expedition loot. I'm not the definitive source on this tho, so feel free to do what you like until a maintainer decides on this.
> 
> > and i could change the cells into 1x2 or 2x2 but that would change the layout of the doors. it isn't MEANT to be a prisoner transport, it's a medical vessel.
> 
> 2x2 would be the very least that's acceptable. (Mostly in the "humane treatment of prisoners" category, we don't want to encourage officers to mistreat prisoners and having good areas for them is part of that.) Alternatively, if you feel this isn't a prisoner transport, I would just remove the holding cells entirely, replacing them with plastitanium walls. Having doubled up walls isn't necessary problematic, especially when there's nothing else you can do with it.

in theory could i just remove the headsets encryption and have it only give common?

**Comment by MagnusCrowe** [comment]
> > > > Oh and also, the glass underneath the thrusters feels very weird. I'd just let those be lattice.
> > > 
> > > 
> > > glass instead of lattice looks cleaner and has been an accepted building practice. it's transparent so you don't have the lattice just sticking out, this example has been see on several other ships including several security vessels. I respect your opinion on the vessel, but i'll add you've been hyper critical of the ship since it's conception. Some people have their preferred building styles and I view these as a more of an artist's signature rather than it being so problematic.
> > 
> > 
> > I agree that glass underneath the thrusters feels very weird, logic wise it should not be a thing and it makes then look disconnected from the shuttle
> > This is something that should be updated and fixed with older ships who were held to different map standards before.
> > Were getting more hyper critical as were getting ships who add more mapper maintainers upkeep while having issues that could have been fixed while they were still being worked on, then having someone else fix later.
> > Look at how many shuttles had to get full atmost redone in the past 2 weeks since this was not a requirement for smalls ships before.
> > There is a baseline doc coming out soon to help streamline this process to have mapping more clear.
> 
> the ship i'm specifically regarding was made by magnus in the last month.

Don't drag me into this, I never put glass under thrusters.  I only use glass under diagonal walls.

**Comment by MagnusCrowe** [comment]
> Right, because a maintainer can do whatever they please and the rest of us get the shaft. The sheer amount of a headache that has become mapping recently has turned me off from wanting to contribute at all. I understand the mapping baselines are in the works at the moment, but don't have one ship be allowed one thing and then change that stance without fixing the lot. It's just getting to the point of if you're a maintainer then anything is free game to you, but when a contributor does something it's met with unfathomable scrutiny and issues. I respect the fact that as a maintainer magnus has a lot more free reign regarding implementation, but when one person follows the example set by said maintainer we get hounded and harassed for it. The fucking glass looks nice, it's transparent and doesn't impact the ship in any form except how you think it should look. Until you figure out what is and isn't acceptable and everyone starts mapping things to that, i think i'll just retire myself from contributing.

First, I was not the one who made the comment about the glass floor under the thrusters (though I do agree, in this circumstance it makes the thrusters look like they are not mounted securely, RP reasoning).  Second, I never put glass under thrusters on the Opportunity.

I get no special treatment, I wasn't even a maintainer when the Opportunity was merged.  

My advice was to use glass under diagonal walls to make them appear more seamless and less blocky.

**Comment by cite2000** [comment]
image
a quick picture of the lattices and glass as an example, i can change this further if need be.

**Comment by TsjipTsjip** [comment]
YML label is suspicious, potentially labeler is broken? Not this PR tho.

**Comment by TsjipTsjip** [comment]
Okay so @cite2000 , during the course of your PR the location for **map** files (the one produced my `savegrid`) was changed to `Resources/Maps/_NF/Shuttles/`, to keep all Frontier stuff in its own folder. Could you move your map file to there? This should fix the labels on this PR.

**Comment by cite2000** [comment]
> YML label is suspicious, potentially labeler is broken? Not this PR tho. (No C# is added when a PR does not change any C# files.)

that's because i did a small change that only changed the price, not the ship itself.

**Comment by cite2000** [comment]
> Okay so @cite2000 , during the course of your PR the location for **map** files (the one produced my `savegrid`) was changed to `Resources/Maps/_NF/Shuttles/`, to keep all Frontier stuff in its own folder. Could you move your map file to there? This should fix the labels on this PR.

that.. isn't an actual path.

**Comment by Cheackraze** [comment]
having the cloner on this ship is a no-go, i suggest turning that room into the singular holding cell if anything. The superpacman cant stay, its not allowed to be a round start buyable generator. Glass under thrusters is also a no-go, and on any other ship that has them should be changed to lattice. The comms issue is because theres just an overabundance of them everywhere because of being essentially double mapped and they are still too often ending up in hands of non-security, we still havent exactly landed on the perfect solution but for now we need to at least trim the free mapped ones.

Im going to say that overall everything @TsjipTsjip has said, I would have also mentioned.

Design-wise, I like the layout, however I feel in some areas your decal work is a bit too noisy.  I feel that the black and red diamond entry way is too noisy and only serves to hide the fact that that room is a bit awkward, empty, and a touch pointless.  You could use that space for some other form or function or at the very least put some chairs down for passengers or a bench or something, and the noisiness of the tiled pattern toned down a bit.

As an idea, I'm not opposed to a medium-sized brig-medic focused ship for the situations you describe in your original post.

**Comment by VividPups** [comment]
@Cheackraze  Intercom was changed.

**Comment by dvir001** [comment]
@cite2000 I updated the PR for you to support the new update, new files location and new naming.
You can keep working on this PR after you run a pull for the changes made here.

Please make sure to use the new ship guide lines when working on this PR:
https://frontierstation14.com/index.php/Ship_Submission_Guidelines

**Comment by Cheackraze** [comment]
if the picture in the post is accurate, this looks good to me @MagnusCrowe if you concur lets get this in

**Comment by MagnusCrowe** [comment]
I'll review it today.

**Review by MagnusCrowe — CHANGES_REQUESTED** [review]
image
Top to bottom:

- Please remove the cigarettes
- Please add vacuum tiles to your corners to prevent space wind from pulling up the glass
- Please move your late join spawn point so it's not on top of the table

After these are done, I will merge it

**Comment by cite2000** [comment]
so i failed to mention this but about a week ago i attempted to pull the changes made and launch it and have been unable to properly launch it since. quite possibly i might need to entirely remake it at this rate.

**Comment by VividPups** [comment]
can you post the error code that you receive when trying to do it?

**Comment by Cheackraze** [comment]
> so i failed to mention this but about a week ago i attempted to pull the changes made and launch it and have been unable to properly launch it since. quite possibly i might need to entirely remake it at this rate.

your work is all saved so far here on github, so in the worst case it would simply be deleting your local repository and pulling it fresh from here. Feel free to reach out on the discord if you're still having troubles

**Comment by cite2000** [comment]
here's a picture of basically what changed, that should be everything that was requested 
image

**Review by MagnusCrowe — APPROVED** [review]
I moved the cryo pump out from under the table, put plating under your grilled windows, replaced the generators with the ship version that runs on map init, and moved the files to the correct mapping folder /_NF/Shuttles/Security/whiskey.yml.

**Pull request merged**
